crates/e2e/tests/e2e/helpers/deploy.rs

```diff
@@ -0,0 +1,124 @@
+use {
+    contracts::{
+        BalancerV2Authorizer,
+        BalancerV2Vault,
+        CoWSwapEthFlow,
+        GPv2AllowListAuthentication,
+        GPv2Settlement,
+        UniswapV2Factory,
+        UniswapV2Router02,
+        WETH9,
+    },
+    ethcontract::{Address, U256},
+    model::DomainSeparator,
+    shared::ethrpc::Web3,
+};
+
+pub struct Contracts {
+    pub balancer_vault: BalancerV2Vault,
+    pub gp_settlement: GPv2Settlement,
+    pub gp_authenticator: GPv2AllowListAuthentication,
+    pub uniswap_v2_factory: UniswapV2Factory,
+    pub uniswap_v2_router: UniswapV2Router02,
+    pub weth: WETH9,
+    pub allowance: Address,
+    pub domain_separator: DomainSeparator,
+    pub ethflow: CoWSwapEthFlow,
+}
+
+impl Contracts {
+    pub async fn deploy(web3: &Web3) -> Self {
+        let network_id = web3.net().version().await.expect("get network ID failed");
+        tracing::info!("connected to test network {}", network_id);
+
+        let accounts: Vec<Address> = web3.eth().accounts().await.expect("get accounts failed");
+        let admin = accounts[0];
+
+        macro_rules! deploy {
```

Comment:
I don't like macros either so I'm fine with removing them. Usually you can write generic functions that only have small overhead over macros. Also fine with keeping them because this PR isn't the one that introduced them.

Reply:
This macro was already there. I can remove it (along with the `tx...` macros), but maybe in another PR?